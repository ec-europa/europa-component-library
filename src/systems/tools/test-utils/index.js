const merge = require('deepmerge');
const twing = require('@ecl-twig/ec-storybook/.storybook/environment');

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) => {
    try {
      const html = twing.render(file, options);
      const test = document.createElement('test');
      test.innerHTML = html.trim();
      resolve(test);
    } catch (error) {
      reject(error);
    }
  });

module.exports = {
  merge,
  renderTwigFileAsNode,
};
