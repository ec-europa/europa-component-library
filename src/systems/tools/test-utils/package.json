{
  "name": "@ecl-twig/test-utils",
  "version": "2.31.1",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
