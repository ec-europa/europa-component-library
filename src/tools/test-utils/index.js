const merge = require('deepmerge');
const twing = require('@ecl/twig-ec-storybook/.storybook/environment');

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) => {
    try {
      const html = twing.render(file, options);
      const div = document.createElement('div');
      div.innerHTML = html.trim();
      resolve(div.firstChild);
    } catch (error) {
      reject(error);
    }
  });

module.exports = {
  merge,
  renderTwigFileAsNode,
};
