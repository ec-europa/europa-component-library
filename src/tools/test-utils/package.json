{
  "name": "@ecl/test-utils",
  "version": "3.4.2",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
