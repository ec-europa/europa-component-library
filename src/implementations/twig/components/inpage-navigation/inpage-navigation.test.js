/**
 * @jest-environment jsdom
 */

import { merge, renderTwigFileAsNode } from '@ecl/test-utils';

import demoData from '@ecl/specs-component-inpage-navigation/demo/data';

describe('Inpage navigation', () => {
  const template = '@ecl/inpage-navigation/inpage-navigation.html.twig';
  const render = (params) => renderTwigFileAsNode(template, params);

  test('renders correctly', () => {
    expect.assertions(1);

    return expect(render(demoData)).resolves.toMatchSnapshot();
  });

  test('renders correctly with extra class names', () => {
    expect.assertions(1);

    const optionsWithExtraClasses = merge(demoData, {
      extra_classes: 'custom-class custom-class--test',
    });

    return expect(render(optionsWithExtraClasses)).resolves.toMatchSnapshot();
  });

  test('renders correctly with extra attributes', () => {
    expect.assertions(1);

    const optionsWithExtraClasses = merge(demoData, {
      extra_attributes: [
        { name: 'data-test', value: 'data-test-value' },
        { name: 'data-test-1', value: 'data-test-value-1' },
      ],
    });

    return expect(render(optionsWithExtraClasses)).resolves.toMatchSnapshot();
  });
});
