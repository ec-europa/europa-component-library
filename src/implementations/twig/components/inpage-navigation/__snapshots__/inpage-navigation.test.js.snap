// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inpage navigation renders correctly 1`] = `
<jest>
  <nav
    aria-labelledby="ecl-inpage-navigation-default"
    class="ecl-inpage-navigation"
    data-ecl-auto-init="InpageNavigation"
    data-ecl-inpage-navigation="true"
  >
    <div
      class="ecl-inpage-navigation__title"
      id="ecl-inpage-navigation-default"
    >
      Page contents
    </div>
    <div
      class="ecl-inpage-navigation__body"
    >
      <div
        class="ecl-inpage-navigation__trigger-wrapper"
      >
        <button
          aria-controls="ecl-inpage-navigation-list"
          aria-expanded="false"
          class="ecl-inpage-navigation__trigger"
          data-ecl-inpage-navigation-trigger="true"
          id="ecl-inpage-navigation-default-trigger"
          type="button"
        >
          <span
            class="ecl-inpage-navigation__trigger-current"
            data-ecl-inpage-navigation-trigger-current="true"
          >
            Page contents
          </span>
          <svg
            aria-hidden="false"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-180 ecl-inpage-navigation__trigger-icon"
            focusable="false"
            role="img"
          >
            <title>
              Show full page table of contents
            </title>
            <use
              xlink:href="/icons.svg#corner-arrow"
            />
          </svg>
        </button>
      </div>
      <ul
        class="ecl-inpage-navigation__list"
        data-ecl-inpage-navigation-list="true"
        id="ecl-inpage-navigation-default-list"
      >
        <li
          class="ecl-inpage-navigation__item"
        >
          <a
            class="ecl-link ecl-inpage-navigation__link"
            data-ecl-inpage-navigation-link=""
            href="#inline-nav-1"
          >
            Heading 1
          </a>
        </li>
        <li
          class="ecl-inpage-navigation__item"
        >
          <a
            class="ecl-link ecl-inpage-navigation__link"
            data-ecl-inpage-navigation-link=""
            href="#inline-nav-2"
          >
            Heading 2 with a long title going on several lines
          </a>
        </li>
        <li
          class="ecl-inpage-navigation__item"
        >
          <a
            class="ecl-link ecl-inpage-navigation__link"
            data-ecl-inpage-navigation-link=""
            href="#inline-nav-3"
          >
            Heading 3
          </a>
        </li>
        <li
          class="ecl-inpage-navigation__item"
        >
          <a
            class="ecl-link ecl-inpage-navigation__link"
            data-ecl-inpage-navigation-link=""
            href="#inline-nav-4"
          >
            Heading 4
          </a>
        </li>
      </ul>
    </div>
  </nav>
</jest>
`;

exports[`Inpage navigation renders correctly with extra attributes 1`] = `
<jest>
  <nav
    aria-labelledby="ecl-inpage-navigation-default"
    class="ecl-inpage-navigation"
    data-ecl-auto-init="InpageNavigation"
    data-ecl-inpage-navigation="true"
    data-test="data-test-value"
    data-test-1="data-test-value-1"
  >
    <div
      class="ecl-inpage-navigation__title"
      id="ecl-inpage-navigation-default"
    >
      Page contents
    </div>
    <div
      class="ecl-inpage-navigation__body"
    >
      <div
        class="ecl-inpage-navigation__trigger-wrapper"
      >
        <button
          aria-controls="ecl-inpage-navigation-list"
          aria-expanded="false"
          class="ecl-inpage-navigation__trigger"
          data-ecl-inpage-navigation-trigger="true"
          id="ecl-inpage-navigation-default-trigger"
          type="button"
        >
          <span
            class="ecl-inpage-navigation__trigger-current"
            data-ecl-inpage-navigation-trigger-current="true"
          >
            Page contents
          </span>
          <svg
            aria-hidden="false"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-180 ecl-inpage-navigation__trigger-icon"
            focusable="false"
            role="img"
          >
            <title>
              Show full page table of contents
            </title>
            <use
              xlink:href="/icons.svg#corner-arrow"
            />
          </svg>
        </button>
      </div>
      <ul
        class="ecl-inpage-navigation__list"
        data-ecl-inpage-navigation-list="true"
        id="ecl-inpage-navigation-default-list"
      >
        <li
          class="ecl-inpage-navigation__item"
        >
          <a
            class="ecl-link ecl-inpage-navigation__link"
            data-ecl-inpage-navigation-link=""
            href="#inline-nav-1"
          >
            Heading 1
          </a>
        </li>
        <li
          class="ecl-inpage-navigation__item"
        >
          <a
            class="ecl-link ecl-inpage-navigation__link"
            data-ecl-inpage-navigation-link=""
            href="#inline-nav-2"
          >
            Heading 2 with a long title going on several lines
          </a>
        </li>
        <li
          class="ecl-inpage-navigation__item"
        >
          <a
            class="ecl-link ecl-inpage-navigation__link"
            data-ecl-inpage-navigation-link=""
            href="#inline-nav-3"
          >
            Heading 3
          </a>
        </li>
        <li
          class="ecl-inpage-navigation__item"
        >
          <a
            class="ecl-link ecl-inpage-navigation__link"
            data-ecl-inpage-navigation-link=""
            href="#inline-nav-4"
          >
            Heading 4
          </a>
        </li>
      </ul>
    </div>
  </nav>
</jest>
`;

exports[`Inpage navigation renders correctly with extra class names 1`] = `
<jest>
  <nav
    aria-labelledby="ecl-inpage-navigation-default"
    class="ecl-inpage-navigation custom-class custom-class--test"
    data-ecl-auto-init="InpageNavigation"
    data-ecl-inpage-navigation="true"
  >
    <div
      class="ecl-inpage-navigation__title"
      id="ecl-inpage-navigation-default"
    >
      Page contents
    </div>
    <div
      class="ecl-inpage-navigation__body"
    >
      <div
        class="ecl-inpage-navigation__trigger-wrapper"
      >
        <button
          aria-controls="ecl-inpage-navigation-list"
          aria-expanded="false"
          class="ecl-inpage-navigation__trigger"
          data-ecl-inpage-navigation-trigger="true"
          id="ecl-inpage-navigation-default-trigger"
          type="button"
        >
          <span
            class="ecl-inpage-navigation__trigger-current"
            data-ecl-inpage-navigation-trigger-current="true"
          >
            Page contents
          </span>
          <svg
            aria-hidden="false"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-180 ecl-inpage-navigation__trigger-icon"
            focusable="false"
            role="img"
          >
            <title>
              Show full page table of contents
            </title>
            <use
              xlink:href="/icons.svg#corner-arrow"
            />
          </svg>
        </button>
      </div>
      <ul
        class="ecl-inpage-navigation__list"
        data-ecl-inpage-navigation-list="true"
        id="ecl-inpage-navigation-default-list"
      >
        <li
          class="ecl-inpage-navigation__item"
        >
          <a
            class="ecl-link ecl-inpage-navigation__link"
            data-ecl-inpage-navigation-link=""
            href="#inline-nav-1"
          >
            Heading 1
          </a>
        </li>
        <li
          class="ecl-inpage-navigation__item"
        >
          <a
            class="ecl-link ecl-inpage-navigation__link"
            data-ecl-inpage-navigation-link=""
            href="#inline-nav-2"
          >
            Heading 2 with a long title going on several lines
          </a>
        </li>
        <li
          class="ecl-inpage-navigation__item"
        >
          <a
            class="ecl-link ecl-inpage-navigation__link"
            data-ecl-inpage-navigation-link=""
            href="#inline-nav-3"
          >
            Heading 3
          </a>
        </li>
        <li
          class="ecl-inpage-navigation__item"
        >
          <a
            class="ecl-link ecl-inpage-navigation__link"
            data-ecl-inpage-navigation-link=""
            href="#inline-nav-4"
          >
            Heading 4
          </a>
        </li>
      </ul>
    </div>
  </nav>
</jest>
`;
